File: app/lib/shared/enums/driver_status.dart
enum DriverStatus {
  pending,
  approved,
  rejected;

  String get value => name;

  static DriverStatus? fromString(String? value) {
    if (value == null) return null;
    for (final status in values) {
      if (status.name == value) return status;
    }
    return null;
  }
}

File: backend/app/Shared/Enums/DriverStatus.php
<?php

namespace App\Shared\Enums;

enum DriverStatus: string
{
    case PENDING = 'pending';
    case APPROVED = 'approved';
    case REJECTED = 'rejected';

    public function label(): string
    {
        return match ($this) {
            self::PENDING => 'Pendiente',
            self::APPROVED => 'Aprobado',
            self::REJECTED => 'Rechazado',
        };
    }

    public static function values(): array
    {
        return array_column(self::cases(), 'value');
    }
}
